superset/charts/client_processing.py

```diff
@@ -340,7 +341,16 @@ def apply_client_processing(  # noqa: C901
         if query["result_format"] == ChartDataResultFormat.JSON:
             df = pd.DataFrame.from_dict(data)
         elif query["result_format"] == ChartDataResultFormat.CSV:
-            df = pd.read_csv(StringIO(data))
+            # Use custom NA values configuration for
+            # reports to avoid unwanted conversions
+            # This allows users to control which values should be treated as null/NA
+            na_values = current_app.config.get("REPORTS_CSV_NA_NAMES", None)
+            if na_values is not None:
+                df = pd.read_csv(
+                    StringIO(data), keep_default_na=False, na_values=na_values
+                )
+            else:
+                df = pd.read_csv(StringIO(data))
```

Comment:
Small nit: `dict.get()` already defaults to `None` when the key doesn't exist, so no need for the second argument. But also, since we know the key exists in config we can just access it using brackets.

With this, we can simplify the call a little bit;

```suggestion
            na_values = current_app.config["REPORTS_CSV_NA_NAMES"]
            df = pd.read_csv(
                StringIO(data),
                keep_default_na=na_value is None,
                na_values=na_values,
            )
```